### PR TITLE
Restructure crate

### LINKED ISSUE #34: Project Restructuring: Move LSPS implementations to `lspsX` modules
As discussed out-of-band, we'll move to a `crate::lspsX` module format afterall.

We should hence restructure the project accordingly once #20 (and possibly follow-ups) lands.

### PR DESCRIPTION
**Pull request by tnull**
Fixes #34 

The structure of this crate was up for discussion right from the beginning (see #1) and we tentatively chose a semantic structure with modules named after their purpose (`transport`, `channel_request`, `jit_channel`). 

However, given that the specifications now follow a "no versioning" model in which breaking changes will result in a new LSPSX standard this structure doesn't really make sense anymore. In particular when LSPS4 will become a reality it live side-by-side with LSPS2, so we will have two `jit_channel` standards for a while. 

To this end, we rename the modules accordingly and `cfg`-gate LSPS1 for now as it's not usable yet, although now checked in CI to make sure we won't break compatibility with it.

### CODE REVIEW

File: src/lsps2/channel_manager.rs [hunk 1]
```diff
@@ -662,21 +659,21 @@ where
 							peer_by_scid.insert(scid, counterparty_node_id);
 						}
 
-						let outbound_jit_channel = OutboundJITChannel::new(
+						let outbound_lsps2 = OutboundJITChannel::new(
```
Comment (johncantrell97) [inline src/lsps2/channel_manager.rs]: hm this feels awkward, also OutboundJITChannel still exists.  Maybe we can keep jit around in some cases as long as it's prefixed by lsps2?

let outbound_lsps2_jit_channel = OutboundLSPS2JITChannel::new()

or let outbound_lsps2_channel = ...

File: src/lsps2/channel_manager.rs [hunk 2]
```diff
@@ -704,9 +701,9 @@ where
 			match outer_state_lock.get(counterparty_node_id) {
 				Some(inner_state_lock) => {
 					let mut peer_state = inner_state_lock.lock().unwrap();
-					if let Some(jit_channel) = peer_state.outbound_channels_by_scid.get_mut(&scid) {
+					if let Some(lsps2) = peer_state.outbound_channels_by_scid.get_mut(&scid) {
```
Comment (johncantrell97) [inline src/lsps2/channel_manager.rs]: same as above, weird to lose channel as part of the name?

File: src/lsps2/channel_manager.rs (hunk 1 above)
Comment (tnull) [reply src/lsps2/channel_manager.rs]: Ah, good point, seems my find/replace was a bit overeager and I didn't catch all the cases. Now reverted most of them in `channel_manager.rs` back to `jit_channel`.

File: src/lsps2/channel_manager.rs (hunk 2 above)
Comment (tnull) [reply src/lsps2/channel_manager.rs]: Yeah, see above.

**Review by johncantrell97 — APPROVED** [review]

**Pull request merged**